pennylane/interfaces/torch.py

```diff
@@ -182,32 +203,13 @@ def backward(ctx, grad_output):  # pragma: no cover
             # subtleties in the torch.autograd.FunctionMeta metaclass, specifically
             # the way in which the backward class is created on the fly
 
-            # determine the QNode variables which should be differentiated
-            diff_indices = None
-            non_diff_indices = set()
-
-            for arg, arg_variable in zip(ctx.saved_tensors, qnode.arg_vars):
-                if not getattr(arg, "requires_grad", False):
-                    indices = [i.idx for i in _flatten(arg_variable)]
-                    non_diff_indices.update(indices)
-
-            if non_diff_indices:
-                diff_indices = set(range(qnode.num_variables)) - non_diff_indices
-
             # evaluate the Jacobian matrix of the QNode
-            jacobian = qnode.jacobian(ctx.args, ctx.kwargs, wrt=diff_indices)
+            jacobian = qnode.jacobian(ctx.args, ctx.kwargs)
             jacobian = torch.as_tensor(jacobian, dtype=grad_output.dtype)
 
             vjp = torch.transpose(grad_output.view(-1, 1), 0, 1) @ jacobian
             vjp = vjp.flatten()
 
-            if non_diff_indices:
-                # Torch requires we return a gradient of size (num_variables,)
-                res = torch.zeros([qnode.num_variables], dtype=grad_output.dtype)
-                indices = np.fromiter(diff_indices, dtype=np.int64)
-                res[indices] = vjp
-                vjp = res
-
```

Comment:
Same questoin: How come this conversion is no longer needed?

Reply:
See https://github.com/XanaduAI/pennylane/pull/660/files#r433597583